src/TimeManager.jl

```diff
@@ -59,123 +50,82 @@ datetime_to_strdate(datetime::Dates.DateTime) =
-"""
-struct Monthly <: AbstractFrequency end
+Convert a string to a `Dates.Period` object.
 
-"""
-    EveryTimestep
-A concrete type for the every-timestep frequency of the callback function.
-"""
-struct EveryTimestep <: AbstractFrequency end
-
-"""
-    trigger_callback(cs, ::Monthly)
+The string can be in one of two formats:
 
-Returns `true` if the current date is equal to or exceeds the saved first of the month at time of 00:00:00.
+1. **Months:** `<NUM>months`, where `<NUM>` is an integer representing the number of months. For example, `"2months"` represents 2 months.
+2. **Other units:** `<NUM><unit>`, where `<NUM>` is a number (integer or floating point) and `<unit>` is one of `secs`, `mins`, `hours`, or `days`.
+   This format is handled by the `time_to_seconds` function and converted to a `Dates.Millisecond` object.
 
 # Arguments
-- `cs`: [CoupledSimulation] containing info about the simulation
-"""
-trigger_callback(cs::Interfacer.CoupledSimulation, ::Monthly) = cs.dates.date[1] >= cs.dates.date1[1] ? true : false
-
-"""
-    CouplerCallback
-
-This is an abstract type for ClimaCoupler's callback functions.
-"""
-abstract type CouplerCallback end
-
-"""
-    do_nothing(::Interfacer.CoupledSimulation, _)
-
-This is a helper callback function that does nothing.
-"""
-do_nothing(::Interfacer.CoupledSimulation, _) = nothing
-
-"""
-    HourlyCallback{FT}
-
-This is a callback type that triggers at intervals of 1h or multiple hours.
-
-# Fields
-
-- `dt`
-- `func`
-- `ref_date`
-- `active`
-- `data
-"""
-@kwdef struct HourlyCallback{FT} <: CouplerCallback
-    dt::FT = FT(1) # hours
-    func::Function = do_nothing
-    ref_date::Array = [Dates.DateTime(0)]
-    active::Bool = false
-    data::Array = []
+- `s::String`: The string to convert to a `Dates.Period`.
+
+# Returns
+- A `Dates.Period` object representing the time period specified in the string.
+
+# Examples
+```julia
+julia> time_to_period("2months")
+2 months
+
+julia> time_to_period("10secs")
+10000 milliseconds
+
+julia> time_to_period("2.5hours")
+9000000 milliseconds
+```
+"""
+function time_to_period(s::String)
+    if occursin("months", s)
+        months = match(r"^(\d+)months$", s)
+        isnothing(months) && error("$(s) has to be of the form <NUM>months, e.g. 2months for 2 months")
+        return Dates.Month(parse(Int, first(months)))
+    else
+        # Milliseconds to support fractional seconds
+        return Dates.Millisecond(1000 * time_to_seconds(s))
+    end
 end
 
 """
-    MonthlyCallback{FT}
+    Callback
 
-This is a callback type that triggers at intervals of 1 month or multiple months.
+A small struct containing a schedule, and the function to be executed on the
+schedule is true.
 
-# Fields
+A `schedule` is a callable object (ie, a function) that takes an integrator-type
+of object and returns true or false.
+TODO: If `cs` contained the correct time, we could just pass `cs` to the schedule
```

Comment:
I recently removed time from `CoupledSimulation` because it wasn't being used or correctly updated, but if it would be useful we can add it back.

Reply:
It could be, but we can do this later